eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java

```diff
@@ -0,0 +1,245 @@
+                                                                 SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH,
+                                                                 VALIDATED);
+        DbsnpSubmittedVariantEntity submittedVariantEntity = new DbsnpSubmittedVariantEntity(
+                SUBMITTED_VARIANT, hashingFunctionSubmitted.apply(submittedVariant), submittedVariant);
+
+        ClusteredVariant clusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig", START_1,
+                                                                 VARIANT_TYPE, VALIDATED);
+        DbsnpClusteredVariantEntity clusteredVariantEntity = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION, hashingFunctionClustered.apply(clusteredVariant), clusteredVariant);
+
+        DbsnpVariantsWrapper wrapper = new DbsnpVariantsWrapper();
+        wrapper.setSubmittedVariants(Collections.singletonList(submittedVariantEntity));
+        wrapper.setClusteredVariant(clusteredVariantEntity);
+        return wrapper;
+    }
+
+    private void assertRsWasStored(DbsnpVariantsWrapper wrapper) {
+        List<DbsnpClusteredVariantEntity> rsEntities = mongoTemplate.find(new Query(),
+                                                                          DbsnpClusteredVariantEntity.class);
+        assertEquals(1, rsEntities.size());
+        assertEquals(wrapper.getClusteredVariant(), rsEntities.get(0));
+    }
+
+    @Test
+    public void writeComplexVariant() throws Exception {
+        SubmittedVariant submittedVariant_1 = new SubmittedVariant("assembly", TAXONOMY_1, "project", "contig", START_1,
+                                                                 "reference", "alternate", CLUSTERED_VARIANT,
+                                                                 SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH,
+                                                                 VALIDATED);
+        SubmittedVariant submittedVariant_2 = new SubmittedVariant("assembly", TAXONOMY_1, "project", "contig", START_1,
+                                                                 "reference", "alternate_2", CLUSTERED_VARIANT,
+                                                                 SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH,
+                                                                 VALIDATED);
+        DbsnpVariantsWrapper wrapper = buildSimpleWrapper();
+        wrapper.setSubmittedVariants(Arrays.asList(
+                new DbsnpSubmittedVariantEntity(SUBMITTED_VARIANT,
+                                                hashingFunctionSubmitted.apply(submittedVariant_1),
+                                                submittedVariant_1),
+                new DbsnpSubmittedVariantEntity(SUBMITTED_VARIANT,
+                                                hashingFunctionSubmitted.apply(submittedVariant_2),
+                                                submittedVariant_2)));
```

Comment:
This is not a clean way to constructor the test object, because it sets with certain values and then modifies it. It gets even worse in `repeatedClusteredVariants` because it initialized multiple objects in different places.

If needed, create another private method that receives the submitted variants you want to initialize the wrapper with.